### PR TITLE
fix(Profile): ContactsView layout and scrolling issues

### LINKED ISSUE #5876: Contact list isn't shown entirely due to restricted viewport
In Settings -> Messaging -> Contacts, ... etc I see a contact list that looks like this:

Screenshot from 2022-05-25 09-46-45

First issue here is that there's no profile picture/identity letters (this is already logged at https://github.com/status-im/status-desktop/issues/5514).

However, what was throwing me off was that I knew I had more contacts than that, but I couldn't see them. It looks like the end of the list was reached. Turns out the other contacts are all there, just hidden in the viewport because of the available scrollable area. Inside that list, reveals the other contacts:
Screenshot from 2022-05-25 09-46-59

**We should**:

1. Make sure the entire vertical space is used to display the items in that view
2. If there's more items than visible, have some indication that this is a scrollable area (maybe with an inset dropshadow at the bottom of the list /cc'ing @John-44 for input)

Regarding 2): the same issue arises when one has a lot of 1-on-1 chats or community channels so might be good to come up with a generic solution we can use in other places.

### LINKED ISSUE #5926: The titles 'Sent' and 'Received' are scrolled with the contact requests lists
# Bug Report

### Additional Information
image

Status desktop version: [Build #2137 (May 30, 2022, 9:02:25 AM)](https://ci.status.im/job/status-desktop/job/branches/job/macos/job/master/2137/)
Operating System: macOS Monterey 12.3 Beta (21E5227a) M1

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #5876
Closes #5926 

### What does the PR do
Fixes ContactsView in profile settings was not behaving correctly when it comes to layout and scrolling

### Affected areas
ContactsView

### Screenshot of functionality
https://user-images.githubusercontent.com/31625338/172392221-d0ed16c7-8245-4254-b3e8-5dd9c7c0695f.mov

### CODE REVIEW

**Comment by alexandraB99** [comment]
@PascalPrecht , @endulab please check on this...I don't have many contacts ( :'( ) so the video is recorded with mock data. Let me know if something like this could work!

**Comment by endulab** [comment]
@alexandraB99 Scroll areas in scroll areas has been always confusing to me.
IMHO, the best option in this case is to have one proxy model and one scroll area - one tree view with sections.
But this will require changes in the model/logic.
On the movie you attached we have 2 lists: Identity Verified Contacts and Contacts. The first one is static (it does not have scroll, right ?), the second one is scrolling. What about making the second view also static. And have scroll for both lists. I am attaching a screenshot which describes what I mean:
image

The solution here is much better than what we have now. So, if you don't want to play with it, still worth to merge it. Good job :)

**Comment by alexandraB99** [comment]
> @alexandraB99 Scroll areas in scroll areas has been always confusing to me. IMHO, the best option in this case is to have one proxy model and one scroll area - one tree view with sections. But this will require changes in the model/logic. On the movie you attached we have 2 lists: Identity Verified Contacts and Contacts. The first one is static (it does not have scroll, right ?), the second one is scrolling. What about making the second view also static. And have scroll for both lists. I am attaching a screenshot which describes what I mean: image
> 
> To recap. The solution you propose in this pr is much better than what we have now. So, if you don't want to play with it, still worth to merge it. Good job :)

@endulab in the video attached the first list has only 4 entries so no need to scroll. the second list has more entries than available space and thus it scrolls. the overall content scrolls by default, the lists are different. so we can either increase the overall content's contentHeight to scroll more down or keep what's proposed here.

File: ui/app/AppLayouts/Profile/views/ContactsView.qml [hunk 1]
```diff
@@ -92,131 +92,113 @@ SettingsContentBase {
...
+                        Global.openProfilePopup(publicKey, null, true)
                     }
                 }
 
-                NoFriendsRectangle {
-                    visible: root.contactsStore.myContactsModel.count === 0
-                    //% "You don’t have any contacts yet"
-                    text: qsTrId("you-don-t-have-any-contacts-yet")
+                ContactsListPanel {
+                    id: vrerifiedContacts
```
Comment (endulab) [inline ui/app/AppLayouts/Profile/views/ContactsView.qml:122]: Probably you want "verifiedContacts"

File: ui/app/AppLayouts/Profile/views/ContactsView.qml [hunk 2]
```diff
@@ -92,131 +92,113 @@ SettingsContentBase {
...
-
-                        onOpenChangeNicknamePopup: {
-                            Global.openProfilePopup(publicKey, null, true)
-                        }
+            Column {
+                ContactsListPanel {
+                    id: verifiedContacts
+                    width: parent.width
+                    height: (contactsListHeight < (stackLayout.height/2)) ? contactsListHeight :
+                            (stackLayout.height-vrerifiedContacts.contactsListHeight)
```
Comment (endulab) [inline ui/app/AppLayouts/Profile/views/ContactsView.qml:103]: Probably you want "verifiedContacts"

File: ui/app/AppLayouts/Profile/test.qml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import QtQuick 2.0
```
Comment (endulab) [inline ui/app/AppLayouts/Profile/test.qml:1]: Is this file needed in this pr ?

**Review by endulab — CHANGES_REQUESTED** [review]

**Comment by endulab** [comment]
@alexandraB99 I really like the new video. I added some minor comments to the pr.

File: ui/app/AppLayouts/Profile/views/ContactsView.qml (hunk 2 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Profile/views/ContactsView.qml]: fixed

File: ui/app/AppLayouts/Profile/test.qml (hunk 3 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Profile/test.qml]: not at all, removed! :D

File: ui/app/AppLayouts/Profile/views/ContactsView.qml (hunk 1 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Profile/views/ContactsView.qml]: fixed, updated

**Review by endulab — APPROVED** [review]

**Comment by elina2015** [comment]
@alexandraB99hey, could you check few issues that I found:
1. The Contacts tab doesn't look good when I received a contact request (The double title 'Contacts' appeared):
image
2. The title 'Received' is shown very close to other titles on the Pending requests tab:
image
3. The title 'Sent' is shown very close to other titles after I accepted all contact requests:
image
4. Default images for profiles are not shown on the Contacts tab:
image
5. Pending requests tab is empty when I received a lot of incoming requests
image
6. The Pending requests tab is empty when I accepted all incoming requests but have some sent pending requests ans switching between tabs:
image
7. The Contacts tab looks incorrect when I have a lot of contacts there:
image

**Comment by alexandraB99** [comment]
@elina2015 pushed fixes, so please have a second look. As about the scroll bar is because when the list has more (hidden) items, we show always the list's scrollbar so that the user is aware that there is more. The bigger scrollbar refers to the entire content. see video:
https://user-images.githubusercontent.com/31625338/174332231-88e9dff0-c45f-4f25-a5f6-ca7a300090ed.mov

**Comment by elina2015** [comment]
@alexandraB99 hey, I tested and the scroll for the requests is not scrolled properly. Please check the attached video:
https://user-images.githubusercontent.com/14942081/174704761-56d0bdd7-3866-4419-86ca-1ec5c8ab99da.mov

**Comment by elina2015** [comment]
@alexandraB99 I tested the rebased build and the issue with scrolling is till reproduced. Could you kindly check? https://drive.google.com/file/d/1UazINkuphmnhBbl1lJ9aNYqlXmwZoDsT/view?usp=sharing
My test environment: macOS Monterey 12.3 Beta (21E5227a)

**Comment by alexandraB99** [comment]
@elina2015 pushed some updates! please check again 👯

**Comment by elina2015** [comment]
@alexandraB99 hey, I tested the updates, but unfortunately the issue is till reproduced: https://drive.google.com/file/d/12sIyMqe8TMtLw8OmpnaD4Qqz_7euiPX0/view?usp=sharing

**Comment by alexandraB99** [comment]
> @alexandraB99 hey, I tested the updates, but unfortunately the issue is till reproduced: https://drive.google.com/file/d/12sIyMqe8TMtLw8OmpnaD4Qqz_7euiPX0/view?usp=sharing
@elina2015 thanks for that! pushed another update so please have a second look :)

**Review by elina2015 — APPROVED** [review]
Tested and Approved

**Review by Khushboo-dev-cpp — APPROVED** [review]

**Pull request merged**